server/lib/nl/detection/date.py

```diff
@@ -103,6 +105,15 @@ def parse_date(query: str, ctr: Counters) -> DateClassificationAttributes:
       dates.append(Date(_LAST_YEARS_PREP, year, year_span=1))
       trigger_strings.append(query[match.start():match.end()])
 
+  for pattern in YEARS_AGO_RE:
```

Comment:
This func is complicated enough that I might add sprinkle some comments on what each block is handling?

Reply:
done